workflows/transport/pika_transport.py

```diff
@@ -381,11 +379,13 @@ def _redirect_callback(ch, method, properties, body):
         try:
             self._channel.start_consuming()
         except pika.exceptions.AMQPChannelError as err:
+            self._connected = False
```

Comment:
This will lead to an invalid state - now the transport class _thinks_ it is disconnected, but the connection and channel are still there and left in some error state. If you want to close the connection why not just call 
```suggestion
            self.disconnect()
```